sdk/core/core/inc/az_span.h

```diff
@@ -260,6 +260,34 @@ AZ_NODISCARD az_result az_span_to_uint64(az_span span, uint64_t* out_number);
  */
 AZ_NODISCARD az_result az_span_to_uint32(az_span span, uint32_t* out_number);
 
+/**
+ * @brief az_span_find searches for `target` in `source`, returning an #az_span if it finds it.
+ *
+ * @param[in] source The #az_span containing the ASCII digits to be searched on.
+ * @param[in] target The #az_span containing the token to be searched in `source`.
+ * @param[out] out_span The #az_span pointing to the first occurrence of `target` in `source`, if it is found.
+ * @return An #az_result value indicating the result of the operation.
+ *          #AZ_OK if `target` is found in `source`
+ *          #AZ_ERROR_EOF if `target` is not found in `source`
```

Comment:
AZ_ERROR_ITEM_NOT_FOUND may be a better error for not found. Maybe we should remove "ITEM" from this symbol and just call it AZ_ERROR_NOT_FOUND. Thoughts?

Reply:
Changed.